packages/theme-addresses/theme-addresses.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * CountryProvinceSelector Constructor
+ * @param {Object} countryOptions Country JSON object
+ */
+export function CountryProvinceSelector(countryOptions) {
+  this.countryOptions = countryOptions;
+}
+
+/**
+ * Helper function that builds the province selector
+ */
+function buildProvince (provinceNodeElement, provinces) {
+  var defaultValue = provinceNodeElement.getAttribute('data-default');
+
+  provinces.forEach(function (option) {
+    var optionElement = document.createElement('option');
+    optionElement.value = option[0];
+    optionElement.textContent = option[1];
+
+    provinceNodeElement.appendChild(optionElement);
+  })
+  provinceNodeElement.value = defaultValue;
+}
+
+/**
+ * Helper function that determines if province selector needs to be build
+ */
+function resolveProvinces (countryNodeElement, provinceNodeElement, selectedValue, options) {
+  var selectedOption = countryNodeElement.querySelector('option[value="' + selectedValue +'"]');
+  var provinces = JSON.parse(selectedOption.getAttribute('data-provinces'));
+
+  provinceNodeElement.innerHTML = '';
```

Comment:
This feels more like a `buildProvince()` operation than a `resolveProvinces()` operation. Thoughts?

Reply:
Also, another way of doing it without `.innerHTML` could be `provinceNodeElement.options.length = 0;`

Reply:
How about `resolveProvinces ` => `buildProvince` and `buildProvince` => `buildProvinceOptions`?